rskj-core/src/main/java/co/rsk/peg/utils/MerkleTreeUtils.java

```diff
@@ -0,0 +1,36 @@
+package co.rsk.peg.utils;
+
+import co.rsk.bitcoinj.core.Sha256Hash;
+
+import static co.rsk.bitcoinj.core.Utils.reverseBytes;
+
+public class MerkleTreeUtils {
+    private static MerkleTreeUtils INSTANCE = null;
+
+    public synchronized static MerkleTreeUtils getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new MerkleTreeUtils();
+        }
+
+        return INSTANCE;
+    }
+
+    private MerkleTreeUtils() {}
+
+    /**
+     * Combines two hashes (representing nodes in a merkle tree) to produce a single hash
+     * that would be the parent of these two nodes.
+     *
+     * @param left The left hand side node bytes
+     * @param right The right hand side node bytes
+     * @return
+     */
+    public Sha256Hash combineLeftRight(Sha256Hash left, Sha256Hash right) {
```

Comment:
I think it's better to just leave this function inside MerkleBranch class as its not used, and probably won't be used by other classes. If you prefer to leave it in a separated class i would sugest to set it as a static method as the helper class don't have any state, so there is no need for a Singleton

Reply:
It is used in two different classes: `Rskip92MerkleProofValidator` and `MerkleBranch`.

Reply:
Will change to static.